packages/@aws-cdk/cdk/lib/physical-name-generator.ts

```diff
@@ -0,0 +1,81 @@
+import crypto = require('crypto');
+import { IResource } from "./resource";
+import { Stack } from './stack';
+
+export class PhysicalNameGenerator {
```

Comment:
Still unclear why we need *any* logic here instead of just using the uniqueId (which already contains the stack name, BTW since it's the first component of the path). Let's not be smart if we don't need to (unless I am missing something).

Reply:
Can you be more specific here? Do you mean that we don't need the `PhysicalNameGenerator` class at all, and the physical name should be just the logical ID?

Reply:
Can you address my question @eladb ? I think you missed it in your last review. It's quite an important issue for me to understand clearly.

Reply:
I think it’s okay to have the generator but the default implementation can be just uniqueId with no extra fluff.

Reply:
got it. okay, so can we do just `${account}${region.cleanup()}${uniqueName}`

Reply:
I'm sorry, I don't understand this comment... you want to embed the account number and the region in the generated physical name...?

Can you sketch out how should the `PhysicalNameGenerator` class look like, in your mind? Maybe with some example (pseudo)code?

Reply:
I thought you said you wanted the "account and region in there" so I proposed to add it, but your implementation looks ok to me. Just simplify it as much as possible. Provided some ideas.